assembly/src/parsers/io_ops.rs

```diff
@@ -90,14 +113,62 @@ pub fn parse_env(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assemb
 // NON-DETERMINISTIC INPUTS
 // ================================================================================================
 
-/// TODO: implement
-pub fn parse_read(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Appends the number of READ operations specified by the operation's immediate value to the span
+/// block. This pushes the specified number of items from the advice tape onto the stack.
+///
+/// # Errors
+///
+/// Returns an AssemblyError if the instruction is invalid, malformed, missing the required
+/// parameter, or does not match the expected operation. Returns an invalid_param AssemblyError if
+/// the parameter is not a decimal value.
+pub fn parse_read(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
```

Comment:
We should probably rename this to `parce_adv` or something similar. Also, we need to update the `match` expression in the `mod.rs` file to select this parser when the first part of a token is "adv".